src/ship.c

```diff
@@ -926,6 +931,12 @@ static int ship_parse( Ship *temp, xmlNodePtr parent )
       xmlr_strd(node,"fabricator",temp->fabricator);
       xmlr_strd(node,"description",temp->description);
       xmlr_int(node,"rarity",temp->rarity);
+      if (xml_isNode(node,"engine_offset")) {
```

Comment:
Maybe call it `engine_track` to make it more clear it's visual only?

Reply:
I'll revamp this part : there will be a node `track_generators`, containing one or more `track_generator`
Each one will have its offset and colour set (engine on, engine off, afterburner, jumping). And maybe track's thickness and duration ?
I think there should also be `nebula_track_generator` that is only active in the nebula (and maybe deactivate the regular generators in the nebula ?)
By this mean, each of the engines of a ship will generate tracks.

This raises one question: should the trail colour sets be defined in a separate xml ?

Reply:
That does sound good, but I'm worried that it will make it more complicated to add and maintain ships. That said, if possible, as much of the common stuff should be moved into separate xml files. For example, only define the position and type of trail in each ship's xml, and have the color, thickness, duration, etc. get looked up would work. It would also allow defining things like "zalek_large_trail", which could be changed afterwards to include new graphics and such.